java/daikon/dcomp/BuildJDK.java

```diff
@@ -3,375 +3,500 @@
-          if (exists(rt)) {
-            return rt;
-          }
-        }
-      }
-      return jarFilename;
-    }
+  /**
+   * Instrument the class files given on the command line instead of the ones in the Java runtime
+   * library.
+   */
+  @Option("Instrument only the class files given on the command line")
+  public static boolean classfiles = false;
 
-    /** @return true iff libRt exists in the filesystem */
-    protected boolean exists(String libRt) {
-      return new File(libRt).exists();
-    }
-  }
+  /**
+   * Used when debugging to terminate processing if an error occurs. This flag is tested in
+   * DCInstrument.
+   */
+  @Option("Halt if an instrumentation error occurs")
+  public static boolean quit_if_error = false;
 
+  /** Print information about the classes being instrumented. */
   private static boolean verbose = false;
 
-  /** Whether or not to instrument java.lang.Object */
-  private static boolean skip_object = true;
-
+  /** Number of class files processed; used for progress display. */
   private int _numFilesProcessed = 0;
 
-  private static String static_map_fname = "dcomp_jdk_static_map";
+  /** Name of entry in output jar containing the statics map. */
+  private static String static_field_id_fname = "dcomp_jdk_static_field_id";
 
+  /** Names of all the classes processed. */
   private static List<String> all_classes = new ArrayList<>();
 
+  /** Names of all methods that DCInstrument could not process. Should be empty. */
   private static List<String> skipped_methods = new ArrayList<>();
 
-  public static String[] known_skipped_methods = new String[] {
-        /*
-        "sun.rmi.transport.proxy.RMIMasterSocketFactory.createSocket",
-        "sun.awt.X11.XWindowPeer.handleButtonPressRelease",
-        "com.sun.jmx.snmp.daemon.CommunicatorServer.run",
-        "sun.nio.ch.SocketChannelImpl.read0",
-        "sun.nio.ch.SocketChannelImpl.read",
-        "sun.rmi.transport.proxy.RMIMasterSocketFactory.createSocket",
-        "java.nio.channels.SocketChannel.open",
-        "sun.security.provider.PolicyFile.init",
-        "java.io.Console.readPassword",
-        "sun.tools.jps.Jps.main",
-        "sun.net.www.MimeTable.saveAsProperties",
-        "sun.misc.Service.parse",
-        "sun.font.Type1Font.readFile",
-        "sun.misc.Resource.getBytes",
-        "java.util.ServiceLoader.parse",
-        "sun.jkernel.Bundle.loadReceipts",
-        "sun.nio.ch.PipeImpl$Initializer.run",
-        "com.sun.tools.javac.jvm.ClassReader.readInputStream",
-        "com.sun.tools.javac.processing.ServiceProxy.parse",
-        "com.sun.tools.javac.zip.ZipFileIndex$DirectoryEntry.initEntries",
-        "com.sun.tools.javac.zip.ZipFileIndex.readIndex",
-        "com.sun.tools.javac.zip.ZipFileIndex.writeIndex",
-        */
-      };
+  /**
+   * A list of methods known to cause DCInstrument to fail. This is used to remove known problems
+   * from the list of failures displayed at the end of BuildJDK's execution.
```

Comment:
In what format?  (I can see it in the diff, but with an empty list someone reading the code can't know it.)